library/src/main/java/net/vrgsoft/library/LinkCrawler.kt

```diff
@@ -125,90 +126,87 @@ class LinkCrawler {
                     || lowerCase.contains("name=\"description\"")
                     || lowerCase.contains("name='description'"))
-                updateMetaTag(metaTags, "description", separeMetaTagsContent(match))
+                updateMetaTag(metaTags, "description", separateMetaTagsContent(match))
             else if (lowerCase.contains("property=\"og:image\"")
                     || lowerCase.contains("property='og:image'")
                     || lowerCase.contains("name=\"image\"")
                     || lowerCase.contains("name='image'"))
-                updateMetaTag(metaTags, "image", separeMetaTagsContent(match))
+                updateMetaTag(metaTags, "image", separateMetaTagsContent(match))
         }
 
         return metaTags
     }
 
     private fun updateMetaTag(metaTags: MutableMap<String, String>, url: String, value: String?) {
-        if (value != null && value.isNotEmpty()) {
-            metaTags.put(url, value)
+        if (value?.isNotEmpty() == true) {
+            metaTags[url] = value
         }
     }
 
     /**
      * Gets content from metatag
      */
-    private fun separeMetaTagsContent(content: String): String {
+    private fun separateMetaTagsContent(content: String): String {
         return Jsoup.parse(content).getElementsByAttribute("content").attr("content")
     }
 
     private fun crawlCode(content: String): String {
-        var result = ""
-        var resultSpan = ""
-        var resultParagraph = ""
-        var resultDiv = ""
-
-        resultSpan = getTagContent("span", content)
-        resultParagraph = getTagContent("p", content)
-        resultDiv = getTagContent("div", content)
-
-        when {
-            resultParagraph.length > resultSpan.length && resultParagraph.length >= resultDiv.length -> result = resultParagraph
-            resultParagraph.length > resultSpan.length && resultParagraph.length < resultDiv.length -> result = resultDiv
-            else -> result = resultParagraph
+        val resultSpan = getTagContent("span", content)
+        val resultParagraph = getTagContent("p", content)
+        val resultDiv = getTagContent("div", content)
+
+        val result = when {
+            resultParagraph.length > resultSpan.length && resultParagraph.length >= resultDiv.length -> resultParagraph
+            resultParagraph.length > resultSpan.length && resultParagraph.length < resultDiv.length -> resultDiv
+            else -> resultParagraph
         }
 
         return htmlDecode(result)
     }
 
-    private fun cannonicalPage(url: String): String {
-        var url = url
+    private fun canonicalPage(oldUrl: String): String {
+        var newUrl = oldUrl
 
         var cannonical = ""
-        if (url.startsWith(HTTP_PROTOCOL)) {
-            url = url.substring(HTTP_PROTOCOL.length)
-        } else if (url.startsWith(HTTPS_PROTOCOL)) {
-            url = url.substring(HTTPS_PROTOCOL.length)
+        if (newUrl.startsWith(HTTP_PROTOCOL)) {
```

Comment:
Use URLUtil.isHttpUrl() instead

Reply:
@pawlo2102 that's a bit risky unless the `substring` uses below are modified as well, since you can't guarantee that Webkit's methods will match the same thing the current `startsWith` use does